cmd/sync/aws.go

```diff
@@ -96,3 +162,44 @@ func (client *AWSClient) getInstancesOfAutoscalingGroup(group *autoscaling.Group
 
 	return result, nil
 }
+
+// Configuration for AWS Cloud Provider
+
+type awsConfig struct {
+	Region    string
+	Upstreams []awsUpstream
+}
+
+type awsUpstream struct {
+	Name             string
+	AutoscalingGroup string `yaml:"autoscaling_group"`
+	Port             int
+	Kind             string
+}
+
+func validateAWSConfig(cfg *awsConfig) error {
+	if cfg.Region == "" {
+		return fmt.Errorf(errorMsgFormat, "region")
+	}
+
+	if len(cfg.Upstreams) == 0 {
+		return fmt.Errorf("There are no upstreams found in the config file")
+	}
+
+	for _, ups := range cfg.Upstreams {
+		if ups.Name == "" {
+			return fmt.Errorf(upstreamNameErrorMsg)
```

Comment:
This is tricky because when we add more providers we will need to have similar messages for Upstreams. I am going to change it for now, and when we add a new Provider we can check this again.